### PR TITLE
Simplify configuration architecture

### PR DESCRIPTION
**Pull request by MuriloDalRi**
This PR is a refactor with the goal of simplifying the process of configuring a team's Slack notifications.

https://trello.com/c/2ucOpKId/3455-simplify-configuration-architecture-for-seal-dependapanda-3

### CODE REVIEW

File: .github/workflows/morning_seal_prs.yml [hunk 1]
```diff
@@ -25,24 +25,4 @@ jobs:
...
-            govwifi
-            content-interactions-on-platform-govuk
-            navigation-and-homepage-govuk
-            ai-govuk
-          )
-
-          for team in ${teams[*]}; do
-            ./bin/seal_runner.rb $team
-          done
+          ./bin/seal_runner.rb
```
Comment (ChrisBAshton) [inline .github/workflows/morning_seal_prs.yml]: I really thought this was missing an argument 😅 maybe another refactor we can add into this PR as a tack-on commit, to make all the script invocations work the same way:

```yml
          ./bin/seal_runner.rb morning_seal_prs
```

File: lib/seal.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@ def initialize(teams)
   def bark(mode: nil)
     teams.each do |team|
       bark_at(team, mode:)
+      sleep 60 if mode == "dependapanda" && team.dependapanda
```
Comment (ChrisBAshton) [inline lib/seal.rb]: Why the `sleep`?
(In any case, I'd move this to the `when "dependapanda"` condition in `bark_at`).

File: lib/seal.rb [hunk 3]
```diff
@@ -41,4 +42,10 @@ def bark_at(team, mode: nil)
   rescue StandardError => e
     puts "Error barking at team '#{team.name}': #{e.message}"
   end
+  
+  def send_quotes_message(team, mode)
+    if team.quotes_days.map(&:downcase).include?(Date.today.strftime("%A").downcase) && team.public_send("#{mode}")
```
Comment (ChrisBAshton) [inline lib/seal.rb]: Why the addition of the `&& team.public_send`? Everything else (except the `sleep 60`) is a straightforward refactor, but this is a change of behaviour. If it's needed, can it be pulled into its own commit?

File: config/alphagov.yml [hunk 4]
```diff
@@ -3,6 +3,10 @@
 
 common_properties: &common_properties
   security_alerts: true
+  morning_seal_quotes: false
```
Comment (ChrisBAshton) [inline config/alphagov.yml:6]: Can this second commit be fixed up to the first?
I can see why you've split it out - to make it easier to review - but otherwise the `morning_seal_quotes: env["MORNING_SEAL_QUOTES"] == "true" || config["morning_seal_quotes"],` bit in the first commit doesn't really work (as `config["morning_seal_quotes"]` would be undefined at the time of commit).

Also makes it a bit easier to determine whether we've carried over the arrays of team names from the scripts, into this config.

File: spec/seal_spec.rb [hunk 5]
```diff
@@ -10,18 +10,23 @@
       slack_channel: "#lions",
       quotes: ["go lions!"],
       quotes_days: ["Monday", "tuesday", "Wednesday"],
+      morning_seal_quotes: true,
```
Comment (ChrisBAshton) [inline spec/seal_spec.rb:13]: I'd fix this up to the first commit too - every commit should have a passing test suite, as a general rule. So I'm guessing commit 1 and 2 have a broken test suite until this point.

File: ignored_ci_repos.yml [hunk 6]
```diff
@@ -1,3 +1,7 @@
+- ckanext-datagovuk
+- ckanext-spatial
+- datagovuk_find
+- datagovuk_publish
```
Comment (ChrisBAshton) [inline ignored_ci_repos.yml]: Can you explain in the commit message:

1) Why do these repos need to be ignored?
2) Why weren't we ignoring them already?

Thanks!

File: README.md [hunk 7]
```diff
@@ -2,63 +2,34 @@
...
-This is a Slack bot that publishes a team's pull requests, Dependabot updates, security alerts, and inspirational quotes to their Slack Channel. Once provided with the organization name, team names, and respective repositories, it posts messages as various animal characters such as the Seal and Dependapanda.
+This is a Slack bot that publishes a team's pull requests, Dependabot updates, security alerts, and inspirational quotes to their Slack Channel.
 
 ![image](https://github.com/alphagov/seal/blob/main/images/readme/informative.png)
 ![image](https://github.com/alphagov/seal/blob/main/images/readme/angry.png)
 
 ## How to use it?
 
-### Config
+All configuration is in [config/alphagov.yml](https://github.com/alphagov/seal/blob/main/config/alphagov.yml), the process is slightly different depending on wether your team is part of GOV.UK or not.
```
Comment (ChrisBAshton) [inline README.md]:
```suggestion
Add your team's configuration to [config/alphagov.yml](https://github.com/alphagov/seal/blob/main/config/alphagov.yml). The process is slightly different depending on whether or not your team is part of GOV.UK.
```

File: README.md [hunk 8]
```diff
@@ -2,63 +2,34 @@
...
 
-Fork the repo and add/change the config files that relate to your github organisation. For example, the alphagov config file is located at [config/alphagov.yml](config/alphagov.yml) and the config for scheduled daily visits can be found in [.github/workflows](.github/workflows)
+### GOV.UK teams
 
-Include your team's name, repositories, and the Slack channel you want to post to. The Seal doesn't currently support private repos, but you can fork this repo and use your own token with your preferred permissions if that's required.
+Include your team's name and the Slack channel you want to post to, these must match the team name and slack channel specified in the [developer docs](https://docs.publishing.service.gov.uk/repos.html#repos-by-team). Do not add a list of repos, they will be pulled from the developer docs automatically. Private repos are not currently supported.
```
Comment (ChrisBAshton) [inline README.md]:
```suggestion
Include your team's name and the Slack channel you want to post to. These *must match the team name and Slack channel* specified in the [Developer Docs](https://docs.publishing.service.gov.uk/repos.html#repos-by-team). Do not add a list of repos, they will be pulled from the developer docs automatically. Private repos are not currently supported.
```

File: lib/team_builder.rb [hunk 9]
```diff
@@ -44,10 +44,11 @@ def build_single_team(team_name)
   end
 
   def build_all_teams
-    static_config.map do |_, team_config|
+    static_config.map do |key, team_config|
       if team_config["repos"].nil?
         team_config["repos"] = govuk_team_repos(team_config["channel"])
       end
+      team_config["ci_checks"] = is_govuk_team?(key)
```
Comment (ChrisBAshton) [inline lib/team_builder.rb]: Another refactor commit idea - can we get Seal to raise an exception if it encounters a GOV.UK team where the repos have been added to `config/alphagov.yml`?

**Review by ChrisBAshton — COMMENTED** [review]
This is looking so much better ⭐ Thanks Murilo!

File: .github/workflows/morning_seal_prs.yml (hunk 1 above)
Comment (MuriloDalRi) [reply .github/workflows/morning_seal_prs.yml]: Yes, I was thinking about this too. I'll change it.

File: lib/seal.rb (hunk 2 above)
Comment (MuriloDalRi) [reply lib/seal.rb]: That sleep was in the workflow and was intended to prevent rate-limiting, I'll move it down to the `dependapanda` case.

File: lib/seal.rb (hunk 3 above)
Comment (MuriloDalRi) [reply lib/seal.rb]: It's because we now call the Seal with either `morning_seal_quotes` or `afternoon_seal_quotes` and this is a short way to check if the team has enabled the alerts with the same name in their YAML config. I'll add something to the commit message. 

See https://stackoverflow.com/questions/1407451/calling-a-method-from-a-string-with-the-methods-name-in-ruby

File: config/alphagov.yml (hunk 4 above)
Comment (MuriloDalRi) [reply config/alphagov.yml:6]: Fair enough, I thought it would be easier to review with the YAML in a separate commit but you're right.

File: lib/team_builder.rb (hunk 9 above)
Comment (MuriloDalRi) [reply lib/team_builder.rb]: Oh that's an interesting suggestion! I'll look into it!

File: ignored_ci_repos.yml (hunk 6 above)
Comment (MuriloDalRi) [reply ignored_ci_repos.yml]: I've spoken to Chris offline and we've decided to drop this commit.

File: lib/team_builder.rb (hunk 9 above)
Comment (MuriloDalRi) [reply lib/team_builder.rb]: I've added a commit to implement your suggestion.

File: lib/seal.rb (hunk 3 above)
Comment (ChrisBAshton) [reply lib/seal.rb]: Thanks Murilo - I think this could be made a bit more readable (and avoid the engineer having to read the commit message to understand why the `public_send` was added), something like:

```suggestion
    today_is_quote_day = team.quotes_days.map(&:downcase).include?(Date.today.strftime("%A").downcase)
    team_has_enabled_this_mode = team.public_send("#{mode}")
    today_is_quote_day && team_has_enabled_this_mode
```

File: lib/seal.rb [hunk 10]
```diff
@@ -24,14 +24,17 @@ def bark(mode: nil)
...
     message = case mode
-              when "quotes"
-                Message.new(team.quotes.sample) if team.quotes_days.map(&:downcase).include?(Date.today.strftime("%A").downcase)
+              when "morning_seal_quotes", "afternoon_seal_quotes"
+                send_quotes_message(team, mode)
               when "dependapanda"
-                MessageBuilder.new(team, :panda).build
+                if team.dependapanda
+                  MessageBuilder.new(team, :panda).build
+                  sleep 60
```
Comment (ChrisBAshton) [inline lib/seal.rb]: I'd add an inline comment here, e.g.

```suggestion
                  sleep 60 # to prevent rate-limiting
```

File: README.md [hunk 11]
```diff
@@ -2,63 +2,34 @@
...
-- [Dependapanda](https://github.com/alphagov/seal/blob/main/.github/workflows/dependapanda.yml): Runs the Dependapanda bot in the morning, posting about Dependabot pull requests and security information.
-
-The Morning Seal and Dependapanda workflows run every weekday morning, while the Afternoon Seal workflow runs every weekday afternoon.
-
-To customize which bots post to your team channel, add or remove your team name in the workflows above. The team name should correspond to a key in [config/alphagov.yml](https://github.com/alphagov/seal/blob/main/config/alphagov.yml), which should have a `channel` property denoting the name of your team's Slack channel.
-
-The `channel` property should match a channel name in the [list of repos by ownership in the Developer Docs](https://docs.publishing.service.gov.uk/repos.html#repos-by-team), otherwise Dependapanda won't know which repos to notify your channel about.
+- morning_seal_quotes: Morning quotes set by your team
+- morning_seal_prs: Morning alerts about old and recent pull requests by team members
```
Comment (ChrisBAshton) [inline README.md]: Another quick refactor suggestion: I'd rename this `seal_prs` (as there is no 'afternoon' equivalent, and for consistency - it's `dependapanda` rather than `morning_dependapanda`) and move this below `afternoon_seal_quotes` in this list.

File: README.md [hunk 12]
```diff
@@ -67,28 +38,6 @@ You can then run the [GitHub Action](https://github.com/alphagov/seal/actions) s
 
 If you don't want to post to Slack you can add a `DRY: true` env to your workflow and the output will only show in the logs.
 
-### Slack configuration
```
Comment (ChrisBAshton) [inline README.md]: I'd keep this section (especially as https://github.com/alphagov/seal/tree/main/images/emoji still exists). If our Slack instance is ever reset, for example, it would be good to know what needs setting.
Could move it into its own file in `docs/` if you like?

File: lib/team_builder.rb [hunk 13]
```diff
@@ -47,6 +47,8 @@ def build_all_teams
     static_config.map do |key, team_config|
       if team_config["repos"].nil?
         team_config["repos"] = govuk_team_repos(team_config["channel"])
+      else
+        raise "Error: #{key} is a GOV.UK team and shouldn't list repos in ./config/alphagov.yml" if is_govuk_team?(key)
```
Comment (ChrisBAshton) [inline lib/team_builder.rb]: Can we add a test for this?

**Review by ChrisBAshton — COMMENTED** [review]
LGTM ✅ only blocker is the README section removal IMO. Also have a few small suggestions below.

File: lib/team_builder.rb [hunk 14]
```diff
@@ -44,16 +39,15 @@ def build_single_team(team_name)
   end
 
   def build_all_teams
-    static_config.map do |key, team_config|
+    static_config.map do |team_name, team_config|
```
Comment (ChrisBAshton) [inline lib/team_builder.rb:42]: We could fix up `team_name` (rather than `key` to the commit where you first edit this line.

File: spec/team_builder_spec.rb [hunk 15]
```diff
@@ -62,16 +63,26 @@
...
+                 },
                ))
   end
 
   let(:env) { { "SEAL_ORGANISATION" => organisation } }
 
   let(:teams) { TeamBuilder.build(env:) }
 
+  before do
+    allow_any_instance_of(TeamBuilder).to receive(:is_govuk_team?).and_return(false)
```
Comment (ChrisBAshton) [inline spec/team_builder_spec.rb]: This is [deprecated](https://github.com/rspec/rspec-mocks/issues/1423) - is there another way of getting the test working?

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍 two nitpicks below, but non-blocking.
